README.md

```diff
@@ -1,4 +1,223 @@
+```kotlin
+val storeWithSideEffect = coroutineScope.createStore<Int, CalculatorAction>(
+    name = "Calculator",
+    initialState = 0,
+    sideEffects = listOf(sideEffect),
+    reducer = { currentState, newAction ->
+        when (newAction) {
+            is CalculatorAction.Add -> currentState + newAction.value
+            is CalculatorAction.Deduct -> currentState - newAction.value
+            is CalculatorAction.Multiply -> currentState * newAction.value
+            is CalculatorAction.Divide -> currentState / newAction.value
+        }
+    }
+)
+```
+
+When we subscribe to `storeWithSideEffect` and trigger side effect by dispatching right actions -
+network request from side effect will be performed:
+```kotlin
+storeWithSideEffect.subscribe { state -> updateUI(state) }
+storeWithSideEffect.dispatch(CalculatorAction.Deduct(1))
+// Current state will become -1
+storeWithSideEffect.dispatch(CalculatorAction.Add(11))
+// Current state will become 10 and network request will happen
+```
+
+`CoRedux` also provides two simplified implementations of `SideEffect` that you might find useful:
+- `SimpleSideEffect` that produces on one input action either one output action or no action.
+- `CancellableSideEffect` that cancels previously running `Job` and starts a new one if the same type of action is dispatched to the store.
+
+### Implementation details
+
+With default values in `createStore()`, CoRedux starts main reducer coroutine ("manager"), when first `StateReceiver` subscribes to `CoreduxStore` instance.
+"Manager" coroutine is a single source of current state that is defined in local scope of coroutine - this allows to prevent any concurrency problems.
+Then "manager" coroutine itself sequentially:
+- starts all side effects coroutines ("workers"), that listens for incoming actions
+- sets current state to initial state
+- immediately pushes the initial state to the `StateReciever`
+- starts listening for a new action from actions `ReceiveChannel`,
+that are populated by either from `dispatch(new_action)` call or from side effects output actions `SendChannel`.
+
+On each new action from `ReceiveChannel` "manager" coroutine sequentially:
+- triggers `reducer()` to update current state inside "manager" coroutine local scope
+- sends updated current state to all `StateReceiver`s
+- sends input action to all side effects, using `BroadcastChannel`
+
+If side effect emits new action based on input action and current state -
+this action is added as a last one to "manager" coroutine input actions `ReceiveChannel` and
+will be processed after all older actions will go through "manager" coroutine input action loop.
+
+### Logging
+
+`createStore()` has as a parameter list of `LogSink`s - add `LogSink` implementations, if you want to log events.
+
+`createStore()` and "manager" coroutine will automatically produce log events,
+though `SideEffect` implementations, as library does not have control over it, should send them using provided `logger`.
```

Comment:
Incorrect grammar. 

Also, I don't understand what you are trying to say. Either rephrase it or remove it as from my point of view the previous sentence is enough, isn't it?

> `createStore()` has as a parameter list of `LogSink`s - add `LogSink` implementations, if you want to log events.

Reply:
I wanted to say that except `SideEffect` implementation other parts will produce log events automatically.

Reply:
I would rephrase this.